automation/hotswap.sh

```
@@ -5,17 +5,20 @@
 
 set -eu
 
-if [ -z "$1" ]
-    then
-        echo "No arguments supplied. Please provide FIAB name as an argument."
-        exit 1
+# see https://stackoverflow.com/questions/3601515/how-to-check-if-a-variable-is-set-in-bash
+if [ -z ${1+x} ]; then
+      echo "No arguments supplied. Please provide FIAB name as an argument."
+      exit 1
 fi
 
 FIAB=$1
 
-printf "Generating the Leo jar...\n\n"
+printf "Generating the Leo jar. This will fail silently here if you have any compilation errors, or print a success message otherwise\n\n"
 # Example: /Users/qi/workspace/leonardo/http/target/scala-2.12/http-assembly-0.1-437ee4a9-SNAPSHOT.jar
 LEO_JAR_PATH=$(sbt -Dsbt.log.noformat=true "project http" assembly | tail -3 | head -2 | grep -o '/Users[^ ]*')
+
+printf "Jar successfully generated."
```

the build could fail silently before